arcgis-ios-sdk-samples/Display information/Display grid/DisplayGridViewController.swift

```diff
@@ -39,26 +40,27 @@ class DisplayGridViewController: UIViewController, UIAdaptivePresentationControl
         mapView.grid = AGSLatitudeLongitudeGrid()
     }
     
-    var settingsViewController: DisplayGridSettingsViewController?
+    private var popoverViewController: UIViewController?
     
-    func makeSettingsViewController() -> DisplayGridSettingsViewController {
-        guard let viewController = storyboard?.instantiateViewController(withIdentifier: "DisplayGridSettingsViewController") as? DisplayGridSettingsViewController else {
+    private func makeSettingsPopoverViewController() -> UIViewController {
+        guard let navController = storyboard?.instantiateViewController(withIdentifier: "SettingsNavigationController") as? UINavigationController,
+            let viewController = navController.topViewController as? DisplayGridSettingsViewController else {
             fatalError()
         }
         
         viewController.mapView = mapView
-        viewController.modalPresentationStyle = .popover
+        navController.modalPresentationStyle = .popover
         
-        return viewController
+        return navController
     }
     
     @IBAction func showSettings(_ sender: Any) {
-        let settingsViewController: DisplayGridSettingsViewController
-        if let viewController = self.settingsViewController {
+        let settingsViewController: UIViewController
+        if let viewController = popoverViewController {
             settingsViewController = viewController
         } else {
-            settingsViewController = makeSettingsViewController()
-            self.settingsViewController = settingsViewController
+            settingsViewController = makeSettingsPopoverViewController()
+            self.popoverViewController = settingsViewController
         }
         settingsViewController.preferredContentSize = {
```

Comment:
It probably would have been easier to leave everything the way it was and then programmatically create the navigation controller here:

```swift
let navigationController = UINavigationController(rootViewController: settingsViewController)
navigationController.preferredContentSize = {
…
```

Reply:
You're probably right but I think either is fine.